ecosystem/indexer-grpc/indexer-transaction-generator/src/transaction_importer.rs

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::config::TransactionImporterPerNetworkConfig;
+use anyhow::Context;
+use aptos_indexer_grpc_utils::create_data_service_grpc_client;
+use aptos_protos::indexer::v1::GetTransactionsRequest;
+use std::{path::Path, time::Duration};
+
+/// GRPC request metadata key for the token ID.
+const GRPC_API_GATEWAY_API_KEY_HEADER: &str = "authorization";
+const GRPC_REQUEST_NAME_HEADER: &str = "x-request-name";
+const GRPC_REQUEST_NAME_VALUE: &str = "testing-framework";
+const TRANSACTION_STREAM_TIMEOUT_IN_SECS: u64 = 60;
+
+impl TransactionImporterPerNetworkConfig {
+    pub async fn run(&self, output_path: &Path) -> anyhow::Result<()> {
+        let mut client = create_data_service_grpc_client(
+            self.transaction_stream_endpoint.clone(),
+            Some(Duration::from_secs(TRANSACTION_STREAM_TIMEOUT_IN_SECS)),
+        )
+        .await?;
+
+        for (version, output_file) in &self.versions_to_import {
+            let mut request = tonic::Request::new(GetTransactionsRequest {
+                starting_version: Some(*version),
+                transactions_count: Some(1),
+                ..GetTransactionsRequest::default()
+            });
+            request.metadata_mut().insert(
+                GRPC_REQUEST_NAME_HEADER,
+                GRPC_REQUEST_NAME_VALUE.parse().unwrap(),
+            );
+            if let Some(api_key) = &self.api_key {
+                request.metadata_mut().insert(
+                    GRPC_API_GATEWAY_API_KEY_HEADER,
+                    format!("Bearer {}", api_key.clone()).parse().unwrap(),
+                );
+            }
+            let mut stream = client.get_transactions(request).await?.into_inner();
+            while let Some(resp) = stream.message().await.context(format!(
+                "[Transaction Importer] Stream ended unexpected for endpoint {:?}",
+                self.transaction_stream_endpoint
+            ))? {
+                let transaction = resp.transactions.first().context(format!(
+                    "[Transaction Importer] Transaction at version {} is not in response.",
+                    version
+                ))?;
+                let json_string = serde_json::to_string_pretty(transaction).context(
+                    format!("[Transaction Importer] Transaction at version {} failed to serialized to json string.", version))?;
+                let output_path = output_path.join(output_file).with_extension("json");
+                tokio::fs::write(output_path, json_string)
+                    .await
+                    .context(format!(
+                        "[Transaction Importer] Failed to write transaction at version {} to file.",
+                        version
+                    ))?;
```

Comment:
When building CI/CD, how do we think about just diffing (without writing) the files to see if there's any change?

Reply:
just at the lines you pointed out, my vision is to surface this up with a configuration `diff-only`. Then CLI can exit with non-zero status to indicate this is a failure and output diff content to stderr/stdout.

Reply:
added comments for follow-up.

Reply:
can also add it as a linear task for tracking purposes.

Reply:
done tracked here: https://linear.app/aptoslabs/issue/EI-522/local-ci-support-to-auto-generated-transactionsdiffing

Reply:
tracked by EI-522.